Frameworks/Foundation/NSURLProtocol_file.mm

```diff
@@ -46,28 +52,19 @@ + (BOOL)canInitWithRequest:(id)request {
 }
 
 - (id)initWithRequest:(id)request cachedResponse:(id)response client:(id)client {
-    [super initWithRequest:request cachedResponse:response client:client];
-
-    _modes = [[NSMutableArray arrayWithObject:@"kCFRunLoopDefaultMode"] retain];
+    if (self = [super initWithRequest:request cachedResponse:response client:client]) {
+        id url = [_request URL];
+        TraceVerbose(TAG, L"Loading %hs", [[url absoluteString] UTF8String]);
 
-    id url = [_request URL];
-    TraceVerbose(TAG, L"Loading %hs", [[url absoluteString] UTF8String]);
-
-    _path = [[url path] copy];
-    // id host = [NSHost hostWithName:hostName];
+        _path.attach([[url path] copy]);
+    }
 
     return self;
 }
 
 - (id)startLoading {
-    const char* pFilePath = [_path UTF8String];
-
-    fpIn = EbrFopen(pFilePath, "rb");
-    if (!fpIn) {
-        TraceVerbose(TAG, L"Couldn't open %hs", pFilePath);
-    } else {
-        TraceVerbose(TAG, L"Opened %hs", pFilePath);
-    }
+    // TODO 1162: An NSInputStream should be initialized here and run loop events should be used to schedule
+    // the file operations.
```

Comment:
I certainly hope NSURLConnection is up to the task. As well, I wonder if the reference platform short-circuits file URL handling.

It is fully possible to load a file from a file URL without spinning a run loop inbetween.